### PR TITLE
fix: handle migrated config settings

### LINKED ISSUE #663: Click to create function not working
```
definitionsProvider.js? [sm]:154 Uncaught (in promise) TypeError: Cannot read property 'replace' of undefined
    at Object.callback (definitionsProvider.js? [sm]:154)
```

when I try to click and create a method from a XML file.

```
let insertText = atom.config.get('appcelerator-titanium.codeTemplates.jsFunction');
insertText = insertText.replace(/(\${text})/g, text).replace(/\\n/g, '\n');
```

insertText is null. Not looked any further yet :smile:

### PR DESCRIPTION
**Pull request by ewanharris**
Previously this package was appcelerator-titanium, during a botched release before atom shutdown
I renamed it to titanium to try and publish a new module (the backend was giving strange errors)
that indicated updates couldn't be published. I should have followed this through with correcting
the lookups for config settings and other things, so this is finally doing that.

It attempts to do it in a backwards compatible manner, first we lookup the new config setting, and
then fallback to the older setting if we can't find it. However, I'm not certain what the behaviour
will be for default values as these will be set in both places

Fixes #663

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -306,4 +306,22 @@ export default {
...
 	},
+
+	/**
+	 * Retrieve a setting value, handling falling back to the old namespace
+	 *
+	 * @param {string} configString - The setting string to retrieve
+	 * @returns {string|boolean|number}
+	 */
+	getConfigSetting (configString) {
+		configString = configString.replace(/^(appcelerator-titanium|titanium)\./, '');
```
Comment (ewanharris) [inline lib/utils.js:317]: Handling `appcelerator-titanium` shouldn't be necessary as I think I got every instance of it, but I figured it was best to be safe

File: lib/ui/toolbar.jsx [hunk 2]
```diff
@@ -469,7 +469,7 @@ export default class Toolbar {
 		this.targetOptions = [];
 		this.targetOptions.push({ value: '', text: 'Devices', disabled: true });
 
-		const selectedTarget = atom.config.get(`appcelerator-titanium.general.${this.normalizedAppName}.ios.selectedTarget`);
+		const selectedTarget = Utils.get(`appcelerator-titanium.general.${this.normalizedAppName}.ios.selectedTarget`);
```
Comment (m1ga) [inline lib/ui/toolbar.jsx:472]: are those `appcelerator-titanium` strings still correct?

**Comment by m1ga** [comment]
Thanks for the quick update :+1: 

Having a closer look tomorrow. Couldn't get it to run at the moment
```js
Uncaught (in promise) TypeError: _utils.default.get is not a function
    at Toolbar.buildCommand (/home/miga/.pulsar/packages/appcelerator-titanium/lib/ui/toolbar.jsx:29)
```
in `const buildCommand = Utils.get(`appcelerator-titanium.general.${this.normalizedAppName}.buildCommand`);`

File: lib/ui/toolbar.jsx (hunk 2 above)
Comment (ewanharris) [reply lib/ui/toolbar.jsx:472]: Ha, and here was me thinking I got them all! They'll be handled by the regex in the function. I need to fix the function name anyway so I'll fix these

**Comment by ewanharris** [comment]
Sigh, I renamed the function from .get but didn't rename the usage. Fixing! (and seeing if I can add a test that covers that)

File: spec/definitionsProvider-spec.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	after(async function () {
+		this.timeout(5000);
+		sandbox.restore();
+		await atom.packages.deactivatePackage(path.join(__dirname, '..'));
+	});
+
+	it('Should suggest to generate i18n strins', async function () {
+		Project.isTitaniumApp = true;
+		atom.config.set('titanium.project.defaultI18nLanguage', 'fr');
```
Comment (ewanharris) [inline spec/definitionsProvider-spec.js:48]: This changes the default language config setting, and then runs the definition provider i18n strings which will call the config lookup for the above string. Should exercise the functionality changed here

Comment (m1ga) [reply spec/definitionsProvider-spec.js:48]: small typo in `...i18n strins` a `g` is missing

**Comment by m1ga** [comment]
working fine now. One thing I saw in the devtool console: it's trying to read the map files out of the old appcelerator-titanium folder:
Screenshot_20230223_094145
doesn't break anything but I wonder why this is the case

**Comment by ewanharris** [comment]
I think the filepath might be down to the package on the registry still being appcelerator-titanium so that is what the install location is, not sure on the invalid source maps though, that's produced by build tooling so could be anything.

I think once we publish a new version/rename the repo then it _should_ then be just `titanium` on the registry

File: spec/definitionsProvider-spec.js (hunk 3 above)
Comment (ewanharris) [reply spec/definitionsProvider-spec.js:48]: Fixed!

**Comment by ewanharris** [comment]
@m1ga Has this been working ok for you?

**Comment by m1ga** [comment]
yes, using it for some days now and didn't have any issue so far

**Comment by ewanharris** [comment]
Awesome, I'll go ahead and get it merged!

**Pull request merged**